cpp/open3d/t/geometry/BoundingVolume.cpp

```
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
+    if (translation.GetDtype() != GetDtype()) {
+        PrintWarningMessageFromTensor("translation", translation.GetDtype(),
+                                      GetDtype());
+    }
+
+    const core::Tensor translation_d = translation.To(GetDtype());
+    if (relative) {
+        center_ += translation_d;
+    } else {
+        center_ = translation_d;
+    }
+    return *this;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Rotate(
+        const core::Tensor &rotation,
+        const utility::optional<core::Tensor> &center) {
+    core::AssertTensorDevice(rotation, GetDevice());
+    core::AssertTensorShape(rotation, {3, 3});
+    if (rotation.GetDtype() != GetDtype()) {
+        PrintWarningMessageFromTensor("rotation", rotation.GetDtype(),
+                                      GetDtype());
+    }
+
+    if (!rotation.T().AllClose(rotation.Inverse(), 1e-5, 1e-5)) {
+        utility::LogWarning(
+                "Invalid rotation matrix. Please make sure the rotation "
+                "matrix is orthogonal.");
+        return *this;
+    }
+
+    const core::Tensor rotation_d = rotation.To(GetDtype());
+    rotation_ = rotation_d.Matmul(rotation_);
+    if (center.has_value()) {
+        core::AssertTensorDevice(center.value(), GetDevice());
+        core::AssertTensorShape(center.value(), {3});
+        core::Tensor center_d = center.value();
+        if (center_d.GetDtype() != GetDtype()) {
+            PrintWarningMessageFromTensor("center", center_d.GetDtype(),
+                                          GetDtype());
+            center_d = center_d.To(GetDtype());
+        }
+        center_ = rotation_d.Matmul(center_ - center_d).Flatten() + center_d;
+    }
+
+    return *this;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Transform(
+        const core::Tensor &transformation) {
+    core::AssertTensorDevice(transformation, GetDevice());
+    core::AssertTensorShape(transformation, {4, 4});
+    if (transformation.GetDtype() != GetDtype()) {
```

Skip warning, silently convert.